extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/BeanContainerImpl.java

```diff
@@ -85,26 +87,32 @@ public ManagedContext requestContext() {
      *
      * @param <T> represents the type that this factory can create
      */
-    private final class DefaultInstanceFactory<T> implements BeanContainer.Factory<T> {
+    private static final class DefaultInstanceFactory<T> implements BeanContainer.Factory<T> {
```

Comment:
Does this need to be `static`?

Reply:
It doesn't, but inner classes should usually be static unless they access the fields of the outer class, which is not happening in this case

Reply:
Fair enough 🤷